executor/aggregate.go

```diff
@@ -668,28 +671,28 @@ func (e *HashAggExec) parallelExec(ctx context.Context, chk *chunk.Chunk) error
 	if e.executed {
 		return nil
 	}
-	for !chk.IsFull() {
-		e.finalInputCh <- chk
+	for {
 		result, ok := <-e.finalOutputCh
-		if !ok { // all finalWorkers exited
+		if !ok {
 			e.executed = true
-			if chk.NumRows() > 0 { // but there are some data left
-				return nil
-			}
 			if e.isChildReturnEmpty && e.defaultVal != nil {
 				chk.Append(e.defaultVal, 0, 1)
 			}
-			e.isChildReturnEmpty = false
 			return nil
 		}
 		if result.err != nil {
 			return result.err
 		}
+		chk.Append(result.chk, 0, result.chk.NumRows())
```

Comment:
If `chk.NumRows == 0 && result.chk.NumRows() > chk.RequiredRows()`, we may use `chk.SwapColumn` to avoiding data copy.

Reply:
We don't need to check `result.chk.NumRows() >= chk.RequiredRows()`? I think just check `chk.NumRow() == 0` is enough?

Reply:
After some disscution with @XuHuaiyu and @yangxuanjia . We dicide to return the chunk directly. Ignore the `RequiredRows` from the upper executor.